docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,203 @@
+---
+page_title: "Migration Guide: Transition out of Serverless Instances and Shared-tier clusters"
+---
+
+# Migration Guide: Transition out of Serverless Instances and Shared-tier clusters
+
+The goal of this guide is to help users transition from Serverless Instances (`mongodbatlas_serverless_instance`) and Shared-tier clusters (M2 and M5) to Flex or Dedicated Clusters. 
+
+After March 2025, all Serverless instances and Shared-tier clusters will automatically convert to Flex/Dedicated clusters, resulting in configuration drift in Terraform. If a Serverless instance or Shared-tier cluster does not fit the constraints of a Flex cluster, it will convert into a Dedicated cluster, resulting in downtime and workload disruption. Otherwise, it will convert to a Flex cluster.
+
+You can migrate from Serverless instances and Shared-tier clusters to Flex or Dedicated clusters manually before March 2025. 
+
+**Note:** We recommend waiting until March 2025 for Serverless instances and Shared-tier clusters to autoconvert.
+
+## From M2/M5 clusters to Flex 
+### Pre-Autoconversion Migration Procedure
+
+1. Create a new Flex Cluster directly from your `.tf` file, e.g.:
+
+    ```terraform
+    resource "mongodbatlas_flex_cluster" "flex_cluster" {
+        project_id = var.project_id
+        name       = "flexClusterName"
+        provider_settings = {
+            backing_provider_name = "AWS"
+            region_name           = "US_EAST_1"
+        }
+        termination_protection_enabled = true
+    }
+    ```
+2. Run `terraform apply` to create the new resource.
+3. Migrate data from your Shared-tier cluster to the Flex cluster using `mongodump` and `mongostore`.
+
+    Please see the following guide on how to retrieve data from one cluster and store it in another cluster: [Convert a Serverless Instance to a Dedicated Cluster](https://www.mongodb.com/docs/atlas/tutorial/convert-serverless-to-dedicated/)
+
+    Verify that your data is present within the Flex cluster before proceeding.
+4. Delete the Shared-tier cluster by running a destroy command against it:
+
+    `terraform destroy -target=mongodbatlas_advanced_cluster.<shared-tier-cluster-name>`
+
+ 5. Remove the resource block for the Shared-tier cluster from your `.tf` file.
+
+### Post-Autoconversion Migration Procedure
+
+Given your Shared-tier cluster fits the constraints of a Flex cluster, it will automatically convert to a Flex cluster in Atlas, retaining all data. The following resolves the configuration drift in Terraform:
```

Comment:
```suggestion
Given your Shared-tier cluster fits the constraints of a Flex cluster, it will automatically convert to a Flex cluster in Atlas, retaining all data. 
The following resolves the configuration drift in Terraform and does not affect the underlying cluster infrastructure:
```